### PR TITLE
Update `Http3Connection` existing problems

### PR DESCRIPTION
**Pull request by hueifeng**
1、`_aborted` It is possible to execute synchronously or asynchronously 
2、Deal with some basic problems

### CODE REVIEW

File: src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs [hunk 1]
```diff
@@ -148,11 +148,14 @@ public void Abort(ConnectionAbortedException ex)
...
-            if (_aborted)
+            lock (_protocolSelectionLock)
             {
-                // It's safe to check for timeouts on a dead connection,
-                // but try not to in order to avoid extraneous logs.
-                return;
+                if (_aborted)
+                {
+                    // It's safe to check for timeouts on a dead connection,
+                    // but try not to in order to avoid extraneous logs.
```
Comment (halter73) [inline src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs:156]: Why put this lock here? As the comment says, we're just checking `_aborted` to short-circuit as an optimization, not for correctness.

Comment (hueifeng) [reply src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs]: Sorry, I ignored it, I changed it back to what it was.

**Comment by halter73** [comment]
Is there any functional issue this PR is trying to address?

**Comment by BrennanConroy** [comment]
Closing as there has been no response.

**Pull request closed without merging**